tests/signal-cast-registerer.cpp

```diff
@@ -66,7 +68,56 @@ struct EigenCastRegisterer : public dynamicgraph::SignalCastRegisterer
   }
 };
 
-EigenCastRegisterer myVectorCast;
+template<typename Derived>
+struct EigenCastRegisterer_M : public dynamicgraph::SignalCastRegisterer
+{
+    typedef Matrix bnuMatrix;
+
+    EigenCastRegisterer_M () :
+        SignalCastRegisterer
+        (typeid(bnuMatrix), dispMatrix, castMatrix, traceMatrix)
+    {}
+
+    static boost::any castMatrix (std::istringstream& iss)
+    {
+        bnuMatrix res;
+        iss >> res;
+        return res;
+    }
+
+    static void dispMatrix (const boost::any& object, std::ostream& os)
+    {
+        const bnuMatrix& m = boost::any_cast<bnuMatrix> (object);
+        os << "[ ";
+        for (int i = 0; i < m.rows(); ++i){
+            os << "[ ";
+            for (int j = 0; j < m.cols(); ++j) {
+                os << m(i, j) << " ";
+            }
+            if (i != m.rows()-1){
+                os << "]; ";
+            }
+            else{
+                os << "] ";
+            }
+        }
+        os << " ];" << std::endl;
```

Comment:
Isn't Matrix an Eigen::Matrix ?
If yes, can you replace that code by `os << m << std::endl`. If needed, you can play with the [Eigen IO formatting](https://eigen.tuxfamily.org/dox/structEigen_1_1IOFormat.html).

Reply:
Hmmm, I don't really understand what you're saying here, I juste have to replace every os << something << std::endl by os << m << std::endl ?

Reply:
The two for loops above only display the matrix. My comment is a suggestion to make the code easier to read. And it's a good practice for later.

Reply:
Ooo yeah, you're right ! I was testing all the possibility but I never think about just put the Matrix instead of doing all the test